vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,902 @@
+    reference_date = TRTSDT,
+    source_vars = exprs(ADT)
+  ) %>%
+  mutate(AVISIT = VISIT)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, VISIT, AVISIT, RSDTC, ADT, ADTF, ADY)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from worst to best response, followed by `NE`). The `AVAL` values are not considered in
+the parameter derivations below, and so changing `AVAL` here would not change
+the result of those derivations. However, please note that the ordering of `AVAL` 
+will be used to determine `ANL01FL` in the subsequent step, ensure that the appropriate 
+`mode` is being set in the `admiral::derive_var_extreme_flag()`.
+
+IMWG ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "NE" ~ 8,
+    arg == "sCR" ~ 7,
+    arg == "CR" ~ 6,
+    arg == "VGPR" ~ 5,
+    arg == "PR" ~ 4,
+    arg == "MR" ~ 3,
+    arg == "SD" ~ 2,
+    arg == "PD" ~ 1,
+    TRUE ~ NA_real_
+  )
+}
+
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp_new(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVAL, AVALC)
+)
+```
+
+## Derive Confirmed Overall Response Parameter {#covr}
+
+Confirmation of response require two consecutive readings of applicable disease 
+parameters (biochemical analyses). These need to be two distinct samples and 
+usually is performed on the same day or at next scheduled visit.  
+Bone marrow assessments and imaging do not need to be confirmed. 
+
+If `RS` contains unconfirmed response and confirmation is performed at 
+next scheduled visit we can derive Confirmed Overall Response based on 
+response at subsequent visit.
+
+While deriving confirmed response, there are some rules to follow:
+
+* Until any response category has been confirmed, response = stable disease (`SD`).  
+* Patients will continue in the last confirmed response category until 
+there is confirmation of progression or improvement to a higher response status.  
+* If the result is not confirmed by a second evaluation, the status is `NE`
+and the prior confirmed disease status remains valid.  
+* Once confirmed: date of the initial test is considered as date of response.
+
+Below table provides a summary of the confirmed response status calculation at 
+each time point.
+
+```{r echo=FALSE}
+list_resp <- tribble(
+  ~"Response 1 time point", ~"Response 2 time point",
+  ~"Confirmed response 1 time point",
+  "sCR", "sCR", "sCR",
+  "CR", "sCR/CR", "max (CR, last confirmed status)",
+  "VGPR", "sCR/CR/VGPR", "max(VGPR, last confirmed status)",
+  "PR", "sCR/CR/VGPR/PR", "max(PR, last confirmed status)",
+  "MR", "sCR/CR/VGPR/PR/MR", "max(MR, last confirmed status)",
+  "sCR", "CR", "last confirmed status",
```

Comment:
wondering why for this row be used max(CR, last confirmed status). for eg 1st  post baseline assessment =sCR and 2nd=CR, and there sCR is no confirmed response yet, so intermediate confirmed response for 1st assessment should be CR correct?
all below are confusing to me, we can discuss on the call?

Reply:
Since we do not have confirmation for sCR and this is 1st postbaseline assessment I would assign SD instead of CR. Can CR be considered a confirmation of CR at 1 visit? Please discuss on the call

Reply:
As per the workflow document sCR->CR is CR at the sCR visit when no better response was previously confirmed. but we can discuss in the call as I might be confused.

Reply:
updated